demos/heat_eq_variable_source/setup.py

```diff
@@ -0,0 +1,67 @@
+from firedrake import *
+from firedrake_adjoint import *
+from pyroteus.metric import *
+from pyroteus.recovery import *
+from opt_adapt.opt import get_state
+
+
+def initial_mesh(n):
+    """
+    Construction of initial mesh
+    """
+    return RectangleMesh(5 * 2**n, 5 * 2**n, (-1, 1), (-1, 1))
+
+
+def initial_control(mesh):
+    """
+    The initial of control parameter
+    In this example, control parameter is not in Real space
+    """
+    V = FunctionSpace(mesh, "CG", 1)
+    return Function(V, name="Control")
+
+
+def forward_run(mesh, control, **kwargs):
```

Comment:
It would be good to add an `outfile=None` kwarg and then output to pvd after the PDE solve. Then you can straightforwardly view the solution field at each iteration using Paraview. See `point_discharge2d` for an example of how to do this. It would be good if you could do the same for `heat_eq_constant_source`, too.